src/sdl2/sdl.rs

```diff
@@ -76,121 +125,123 @@ impl Drop for Sdl {
     }
 }
 
-/// A RAII value representing initalized SDL subsystems. See `sdl2::Sdl::init_subsystem()`.
-///
-/// Subsystem initialization is ref-counted. Once `Subsystem::drop()` is called,
-/// the specified subsystems' ref-counts are decremented via `SDL_QuitSubSystem`.
-pub struct Subsystem<'sdl> {
-    flags: u32,
-    _marker: PhantomData<&'sdl Sdl>
+// No subsystem can implement `Send` because the destructor, `SDL_QuitSubSystem`,
+// utilizes non-atomic reference counting and should thus be called on a single thread.
+// Some subsystems have functions designed to be thread-safe, such as adding a timer or accessing
+// the event queue. These subsystems implement `Sync`.
+
+macro_rules! subsystem {
+    ($name:ident, $flag:expr) => (
+        impl $name {
+            #[inline]
+            fn new(sdl: &Sdl) -> SdlResult<$name> {
+                let result = unsafe { ll::SDL_InitSubSystem($flag) };
+
+                if result == 0 {
+                    Ok($name {
+                        _subsystem_drop: Rc::new(SubsystemDrop {
+                            _sdldrop: sdl.sdldrop.clone(),
+                            flag: $flag
+                        })
+                    })
+                } else {
+                    Err(get_error())
+                }
+            }
+        }
+    );
+    ($name:ident, $flag:expr, nosync) => (
+        #[derive(Clone)]
+        pub struct $name {
+            /// Subsystems cannot be moved or (usually) used on non-main threads.
+            /// Luckily, Rc restricts use to the main thread.
+            _subsystem_drop: Rc<SubsystemDrop>
+        }
+
+        subsystem!($name, $flag);
+    );
+    ($name:ident, $flag:expr, sync) => (
+        pub struct $name {
+            /// Subsystems cannot be moved or (usually) used on non-main threads.
+            /// Luckily, Rc restricts use to the main thread.
+            _subsystem_drop: Rc<SubsystemDrop>
+        }
+        unsafe impl Sync for $name {}
```

Comment:
As long as none of the methods clone the `Rc` onto another thread or access the ref-count, it should be safe.